src/core/data/server.c

```diff
@@ -223,7 +268,7 @@ core_server_setup(server_options_st *options, server_metrics_st *metrics)
     }
 
     hdl->accept = (channel_accept_fn)tcp_accept;
-    hdl->reject = (channel_reject_fn)tcp_reject;
+    hdl->reject = (channel_reject_fn)tcp_reject_all;
```

Comment:
Ah you called out something important here- admin thread can interfere with server thread still! So I will replace all the `*_borrow()` with `*_create()` and similarly on the return side so admin thread is not messing with resource pooling. This way we should still attempt to accept the next socket instead of rejecting all, since we are basically trying our luck with `malloc()`, which is different scenario from knowing we won't have anything to borrow until the next return event has arrived.

But even if we don't use `tcp_reject()` in the admin thread, I think it's OK to just keep it. You can see different assumptions of how resources are managed will lead to different reject behavior being more appropriate.

Reply:
I'm not too worried about having too many incoming connections on the admin port and resulting in a multitude of stream objects being created. For the current use (mostly stats) that has negligible effect on the overall RSS.